### PR TITLE
{2023.06}[2023a] Paraver 4.11.4

### PR DESCRIPTION
**Pull request by boegel**
```
6 out of 69 required modules missing:

* Graphene/1.10.8-GCCcore-12.3.0 (Graphene-1.10.8-GCCcore-12.3.0.eb)
* elfutils/0.189-GCCcore-12.3.0 (elfutils-0.189-GCCcore-12.3.0.eb)
* GStreamer/1.22.5-GCC-12.3.0 (GStreamer-1.22.5-GCC-12.3.0.eb)
* GST-plugins-base/1.22.5-GCC-12.3.0 (GST-plugins-base-1.22.5-GCC-12.3.0.eb)
* wxWidgets/3.2.2.1-GCC-12.3.0 (wxWidgets-3.2.2.1-GCC-12.3.0.eb)
* Paraver/4.11.4-GCC-12.3.0 (Paraver-4.11.4-GCC-12.3.0.eb)
```

### CODE REVIEW

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
Building of `GST-plugins-base-1.22.5-GCC-12.3.0.eb` dependency fails with:

```
[146/754] Generating gst-libs/gst/tag/GstTag-1.0.gir with a custom command (wrapped by meson to set env)
FAILED: gst-libs/gst/tag/GstTag-1.0.gir
...
/bin/bash: /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/lib64/libtinfo.so.6: no version information available (required by /bin/bash)
/bin/bash: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/lib64/libtinfo.so.6)
...
ninja: build stopped: subcommand failed.
```

This is a problem that looks very similar to what we ran into in with `at-spi2-core` #428, which was fixed in #454...

**Comment by ocaisa** [comment]
This now requires a sync with the default branch for CI or builds to pass

**Comment by bedroge** [comment]
> Building of `GST-plugins-base-1.22.5-GCC-12.3.0.eb` dependency fails with:
> 
> ```
> [146/754] Generating gst-libs/gst/tag/GstTag-1.0.gir with a custom command (wrapped by meson to set env)
> FAILED: gst-libs/gst/tag/GstTag-1.0.gir
> ...
> /bin/bash: /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/lib64/libtinfo.so.6: no version information available (required by /bin/bash)
> /bin/bash: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/lib64/libtinfo.so.6)
> ...
> ninja: build stopped: subcommand failed.
> ```
> 
> This is a problem that looks very similar to what we ran into in with `at-spi2-core` #428, which was fixed in #454...

Yes, it indeed looks like the same problem. For `at-spi2-core` I've fixed it by adding a `--lib-dirs-envvar=FILTER_LD_LIBRARY_PATH` (which prevents that `$LD_LIBRARY_PATH` is being set) to `gir_extra_args` near the affected call to `gnome.generate_gir`. I see several `meson.build` files in the sources of gst-plugins-base that call `gnome.generate_gir`, though, but all of them set `gir_extra_args = gir_init_section + something_else`. The `gir_init_section` is only defined in the `meson.build` of the top-level dir, so you can probably add the same thing there.

However, since we're now running into the same thing again, we could perhaps also look into patching G-Object-Retrospection itself, to prevent that we have to do this same fix over and over again in several other packages...

**Comment by bedroge** [comment]
By setting `--lib-dirs-envvar` you're basically preventing that `runtime_path_envvar` contains `LD_LIBRARY_PATH` (https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.76.1/giscanner/ccompiler.py?ref_type=tags#L224), and in turn that means that `LD_LIBRARY_PATH` won't be set as an environment variable a bit further down: 
https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.76.1/giscanner/ccompiler.py?ref_type=tags#L267

We could try patching this code somehow, but I'm not sure how we can detect here that we're filtering `LD_LIBRARY_PATH`? Maybe we can check in that if statement if `LD_LIBRARY_PATH` is already empty (i.e. if it's not in `os.environ`), and if so, never set it anyway? Because if we're not filtering it, the GObject-Introspection module itself should already have set `LD_LIBRARY_PATH`.

@boegel thoughts?

**Comment by bedroge** [comment]
This should still be moved to the EB 4.9.2 easystack (and the PR itself still needs to be merged as well), but let's see if https://github.com/EESSI/software-layer/pull/646 also fixes the issues of this PR.

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

File: easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.1-2023a.yml [hunk 1]
```diff
@@ -22,3 +22,6 @@ easyconfigs:
         from-pr: 20784
   - Valgrind-3.21.0-gompi-2023a.eb
   - OrthoFinder-2.5.5-foss-2023a.eb
+  - Paraver-4.11.4-GCC-12.3.0.eb:
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.1-2023a.yml:25]: @boegel Could you move this to the EB 4.9.4 file?

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

File: easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.4-2023a.yml [hunk 2]
```diff
@@ -33,3 +33,6 @@ easyconfigs:
       options:
         # see https://github.com/easybuilders/easybuild-easyblocks/pull/3519
         include-easyblocks-from-commit: e4a3ff1932350d575dffc7597435609fad6dd691
+  - Paraver-4.11.4-GCC-12.3.0.eb:
+      options:
+        from-pr: 20230
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.4-2023a.yml:38]:
```suggestion
        # see https://github.com/easybuilders/easybuild-easyconfigs/pull/20230
        from-commit: 91c8df6b4c0810061e9f325427c9c79e961bc4b0
```

**Review by bedroge — APPROVED** [review]
Lgtm! (the easyconfigs PR wasn't changed anymore, so no need to rebuild this after switching from `from-pr` to `from-commit`)

**Pull request merged**